index.js

```diff
@@ -1,83 +1,144 @@
+  }, options);
+
+  var derivedDefaultCleanOptions = {
+        taskPrefix: options.taskPrefix,
+        dist: options.dist
+      },
+      derivedDefaultBrowserifyOptions = {
+        taskPrefix: options.taskPrefix,
+        root: './' + options.src + '/app',
+        dist: options.dist
+      },
+      derivedDefaultStylesOptions = {
+        taskPrefix: options.taskPrefix,
+        root: options.src + '/app.scss',
+        dist: options.dist
+      },
+      derivedDefaultCopyOptions = {
+        taskPrefix: options.taskPrefix,
+        src: options.src,
+        excludes: 'js|scss',
+        dist: options.dist
+      },
+      cleanOptions, browserifyOptions, stylesOptions, copyOptions,
+      cleanTask, browserifyTask, stylesTask, copyTask,
+      buildTasks = [];
+
+  if (!options.copy.skip) {
+    buildTasks.push(options.taskPrefix + 'copy-resources');
+
+    /* Copy all resources to dist */
+    copyOptions = _.merge({}, derivedDefaultCopyOptions, options.copy);
+    copyTask = copyTaskGenerator(gulp, copyOptions);
+  }
 
-  /* Browserify bundler */
-  watchify.args.debug = ($.util.env.type !== 'production');
-  configureBrowserify = function() {
-    browserifyBundler.add('./' + paths.app + '/' + paths.scripts + '/main');
-    for (var i = 0; i < transformStack.length; i++) {
-      browserifyBundler.transform(transformStack[i]);
+  if (!options.styles.skip) {
+    buildTasks.push(options.taskPrefix + 'build-styles');
+
+    /* Build all styles */
+    stylesOptions = _.merge({}, derivedDefaultStylesOptions, options.styles);
+    stylesTask = stylesTaskGenerator(gulp, stylesOptions);
+  }
+
+  if (!options.browserify.skip) {
+    buildTasks.push(options.taskPrefix + 'build-app');
+
+    /* Reduce all javascript to app.js */
+    browserifyOptions = _.merge({}, derivedDefaultBrowserifyOptions, options.browserify);
+    browserifyTask = browserifyTaskGenerator(gulp, browserifyOptions);
+  }
+
+  if (!options.clean.skip) {
+    /* Clean the options.clean.dist directory */
+    cleanOptions = _.merge({}, derivedDefaultCleanOptions, options.clean);
+    cleanTask = cleanTaskGenerator(gulp, cleanOptions);
+  }
+
+  /* Full build */
+  gulp.task(options.taskPrefix + 'build', function(callback) {
+    var browserifyCompleteFn = function() {
+          log.mark('[BROWSERIFY] complete!');
+          callback();
+        };
+
+    if (!options.browserify.skip) {
+      browserifyTask.createBundler();
     }
-  };
 
-  /* Clean the paths.dist directory */
-  gulp.task('clean', function() {
-    return gulp.src(paths.dist, {read: false})
-      .pipe($.rimraf());
+    if (options.clean.skip) {
+      runSequence(buildTasks, browserifyCompleteFn);
+    } else {
+      runSequence(options.taskPrefix + 'clean', buildTasks, browserifyCompleteFn);
+    }
   });
+  // Alias default to do the build.  After this file is run the default task can be overridden if desired.
+  gulp.task('default', ['build']);
```

Comment:
This is where I expect this file to end. Should the watch task be in it's own file?